kyaml/kio/filters/container.go

```diff
@@ -24,6 +26,106 @@ import (
 // non-zero.
 // The full set of environment variables from the parent process
 // are passed to the container.
+//
+// Function Scoping:
+// ContainerFilter applies the function only to Resources to which it is scoped.
+//
+// Resources are scoped to a function if any of the following are true:
+// - the Resource were read from the same directory as the function config
+// - the Resource were read from a subdirectory of the function config directory
+// - the function config is in a directory named "functions" and
+//   they were read from a subdirectory of "functions" parent
+// - the function config doesn't have a path annotation (considered globally scoped)
+// - the ContainerFilter has GlobalScope == true
+//
+// In Scope Examples:
+//
+// Example 1: deployment.yaml and service.yaml in function.yaml scope
+//            same directory as the function config directory
+//     .
+//     ├── function.yaml
+//     ├── deployment.yaml
+//     └── service.yaml
+//
+// Example 2: apps/deployment.yaml and apps/service.yaml in function.yaml scope
+//            subdirectory of the function config directory
+//     .
+//     ├── function.yaml
+//     └── apps
+//         ├── deployment.yaml
+//         └── service.yaml
+//
+// Example 3: apps/deployment.yaml and apps/service.yaml in functions/function.yaml scope
+//            function config is in a directory named "functions"
+//     .
+//     ├── functions
+//     │   └── function.yaml
+//     └── apps
+//         ├── deployment.yaml
+//         └── service.yaml
+//
+// Out of Scope Examples:
```

Comment:
Thinking about the UX of this. Does the user get any indication (e.g. warning or a verbose debugging behind a flag) when resources are out of scope? Concern here is that in a large config directory, users may struggle with understanding why resources are in scope.

Reply:
Hm, I'd expect not being in scope to be the common case.  Maybe we could develop an explicit scoping file?